File: src/main/java/ch/unifr/digits/webprotege/attestation/public/attestation/ontology-contract.js
function OntologyContract(web3, jsonInterface, address) {
    this.delegate = new web3.eth.Contract(jsonInterface, address);
    this.verify = function(from, ontologyIri, versionIri, hash) {
        return this.delegate.methods.verify(ontologyIri, versionIri, hash).call({from: from});
    }
    this.attest = function(from, ontologyIri, versionIri, name, hash) {
        return this.delegate.methods.attest(ontologyIri, versionIri, name, hash).send({from: from});
    }
}

File: src/main/java/ch/unifr/digits/webprotege/attestation/contract/OntologyContract.java
package ch.unifr.digits.webprotege.attestation.contract;

import ch.unifr.digits.gwt.web3.js.Web3;
import ch.unifr.digits.gwt.web3.js.core.TransactionReceipt;
import elemental2.promise.Promise;
import jsinterop.annotations.JsMethod;
import jsinterop.annotations.JsPackage;
import jsinterop.annotations.JsType;

@JsType(isNative = true, namespace = JsPackage.GLOBAL)
public class OntologyContract {

    public OntologyContract(Web3 web3, Object jsonInterface, String address) {}

    @JsMethod
    public native Promise<VerifyContractReturn> verify(String from, String ontologyIri, String versionIri, String hash);

    @JsMethod
    public native Promise<TransactionReceipt> attest(String from, String ontologyIri, String versionIri, String name, String hash);

}
